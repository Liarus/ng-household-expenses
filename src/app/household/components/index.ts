export * from './household-list';
export * from './household-dialog';
export * from './household-tiles';
export * from './household-search';
