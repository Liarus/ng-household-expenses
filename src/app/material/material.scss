.mat-bottom-sheet-container {
  padding: 0 !important;
}

.mat-table {
  mat-row, mat-header-row, mat-footer-row {
    padding-left: 24px;
    padding-right: 24px;
  }
  
  mat-cell:first-child, mat-footer-cell:first-child, mat-header-cell:first-child {
    padding-left: 0;
  }
  
  mat-cell:last-child, mat-footer-cell:last-child, mat-header-cell:last-child {
    padding-right: 0;
  }
}

mat-expansion-panel {
  mat-expansion-panel-header {
    height: 48px; // height may be different for you
  }
  .mat-expansion-panel-content {
    height: 0;
  }
  &.mat-expanded {
    mat-expansion-panel-header {
      height: 64px; // height may be different for you
    }
    .mat-expansion-panel-content {
      height: auto;
    }
  }
}