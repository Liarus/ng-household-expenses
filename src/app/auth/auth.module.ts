import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { AuthEffects } from './store/effects/auth.effects';
import { reducers } from './store/reducers';
import { AuthRoutingModule } from './auth-routing.module';
import { LoginPageComponent } from './containers';
import { LoginFormComponent } from './components';
import { MaterialModule } from '../material/material.module';

export const COMPONENTS = [
  LoginPageComponent,
  LoginFormComponent
];

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    AuthRoutingModule,
    MaterialModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: COMPONENTS,
})
export class AuthModule {}
