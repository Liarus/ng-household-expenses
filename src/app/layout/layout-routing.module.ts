import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { AuthLoggedInGuard } from '../auth/services/authLoggedIn-guard.service';
import { LayoutPageComponent } from './containers';
import { LoginPageComponent } from '../auth/containers';

const routes: Routes = [
  {
    path: '',
    component: LayoutPageComponent,
    children: [
      {
        path: '',
        redirectTo: 'home',
        pathMatch: 'full'
      },
      {
        path: 'login',
        component: LoginPageComponent,
      },
      {
        path: 'home',
        loadChildren: '../dummy/dummy.module#DummyModule',
        canActivate: [AuthLoggedInGuard]
      },
      {
        path: 'households',
        loadChildren: '../household/household.module#HouseholdModule',
        canActivate: [AuthLoggedInGuard]
      }
    ]
  }
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class LayoutRoutingModule {}
