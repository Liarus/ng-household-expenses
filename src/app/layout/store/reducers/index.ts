import { createFeatureSelector, createSelector } from '@ngrx/store';

import * as fromLayout from './layout.reducer';

export const getLayoutState = createFeatureSelector<fromLayout.State>('layout');

export const getIsSidebarExpanded = createSelector(
  getLayoutState,
  fromLayout.getIsSidebarExpanded
);

export const getWindowWidth = createSelector(
  getLayoutState,
  fromLayout.getWindowWidth
);

export const getWindowHeight = createSelector(
  getLayoutState,
  fromLayout.getWindowHeight
);

export const getMenuItems = createSelector(
  getLayoutState,
  fromLayout.getMenuItems
);

export const getIsMobile = createSelector(
  getLayoutState,
  fromLayout.getIsMobile
);
