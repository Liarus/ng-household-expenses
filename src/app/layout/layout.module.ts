import { CommonModule } from '@angular/common';
import { NgModule} from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { FlexLayoutModule } from '@angular/flex-layout';
import { EffectsModule } from '@ngrx/effects';

import { MaterialModule } from '../material/material.module';
import { ToolbarComponent, SidebarComponent } from './components';
import { LayoutEffects } from './store/effects/layout.effects';
import { reducer } from './store/reducers/layout.reducer';
import { LayoutPageComponent } from './containers';
import { LayoutRoutingModule } from './layout-routing.module';

const COMPONENTS = [
  ToolbarComponent,
  SidebarComponent,
  LayoutPageComponent
];

const PROVIDERS = [
];

@NgModule({
  imports: [
    CommonModule,
    MaterialModule,
    LayoutRoutingModule,
    FlexLayoutModule,
    StoreModule.forFeature('layout', reducer),
    EffectsModule.forFeature([LayoutEffects]),
  ],
  declarations: [
    COMPONENTS
  ]
})
export class LayoutModule {}
