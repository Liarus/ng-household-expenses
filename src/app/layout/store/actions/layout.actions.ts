import { createAction, props } from '@ngrx/store';

import { WindowSize } from '../../models/windowSize.model';
import { MenuItem } from '../../models/menuItem.model';

export const openSidebar = createAction(
  '[Layout] Open Sidebar',
);

export const closeSidebar = createAction(
  '[Layout] Close Sidebar',
);

export const toggleSidebar = createAction(
  '[Layout] Toggle Sidebar',
);

export const windowResized = createAction(
  '[Layout/API] Window Resized',
  props<{ result: WindowSize }>()
);

export const applyMenuItems = createAction(
  '[Layout/API] Apply Menu Items',
  props<{ items: MenuItem[] }>()
);

export const refreshMenuItems = createAction(
  '[Layout/API] Refresh Menu Items',
);
