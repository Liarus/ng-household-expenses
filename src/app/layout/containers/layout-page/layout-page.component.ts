import { Component, HostListener } from '@angular/core';
import { Store, select } from '@ngrx/store';

import * as fromLayout from '../../store/reducers/index';
import * as fromAuth from '../../../auth/store/reducers';
import * as fromRoot from '../../../store/reducers';
import * as LayoutActions from '../../store/actions/layout.actions';
import * as AuthActions from '../../../auth/store/actions/auth.actions';

@Component({
  selector: 'app-layout-page',
  template: `
    <app-toolbar [user]="user$ | async"
      (toggleSidebar)="onSidebarToggled()"
      (logout)="onLogout()"
    ></app-toolbar>
    <app-sidebar [isExpanded]="isSidebarExpanded$ | async"
      [menuItems]="menuItems$ | async"
    >
      <router-outlet></router-outlet>
    </app-sidebar>
  `
})
export class LayoutPageComponent {
  user$ = this.store.pipe(select(fromAuth.getLoggedUser));
  isSidebarExpanded$ = this.store.pipe(select(fromLayout.getIsSidebarExpanded));
  menuItems$ = this.store.pipe(select(fromLayout.getMenuItems));
  title = 'ng-household-expenses';

  constructor(private store: Store<fromRoot.State>) { }

  @HostListener('window:resize', ['$event'])
  onResize($event: any) {
    this.store.dispatch(new LayoutActions.ResizeWindow({
      width: $event.target.innerWidth,
      height: $event.target.innerHeight
    }));
  }

  onSidebarToggled() {
    this.store.dispatch(new LayoutActions.ToggleSidebar());
  }

  onLogout() {
    this.store.dispatch(new AuthActions.Logout());
  }
}
