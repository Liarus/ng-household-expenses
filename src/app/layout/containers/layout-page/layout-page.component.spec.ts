import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { StoreModule, Store } from '@ngrx/store';

import { LayoutPageComponent } from './layout-page.component';
import { MockStore } from '../../../shared/tests/mockStore';
import { MaterialModule } from '../../../material/material.module';
import { SidebarComponent, ToolbarComponent } from '../../components';
import * as LayoutActions from '../../store/actions/layout.actions';
import * as AuthActions from '../../../auth/store/actions/auth.actions';

describe('LayoutPageComponent', () => {
  let component: LayoutPageComponent;
  let fixture: ComponentFixture<LayoutPageComponent>;

  let store: MockStore<any>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        NoopAnimationsModule,
        RouterTestingModule,
        StoreModule.forRoot({})
      ],
      declarations: [
        LayoutPageComponent,
        ToolbarComponent,
        SidebarComponent
      ],
      providers: [
        { provide: Store, useClass: MockStore }
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    store = TestBed.get(Store);
    store.setState({
      layout: {
        menuItems: [
          {
            url: '/1',
            title: 'menuTitle1',
            icon: 'home',
            permissions: ['CanSeeUsers'],
            hidden: false
          },
          {
            url: '/2',
            title: 'menuTitle2',
            icon: 'home',
            permissions: ['CanSeeUsers'],
            hidden: false
          },
          {
            url: '/3',
            title: 'menuTitle3',
            icon: 'home',
            permissions: ['CanSeeUsers'],
            hidden: false
          }
        ],
        isSidebarExpanded: true
      },
      auth: {
        status: {
          user: {
            id: '7bb78f33-0612-409e-a1d6-4341fcee9a7e',
            name: 'UserName'
          }
        }
      }
    });

    fixture = TestBed.createComponent(LayoutPageComponent);
    component = fixture.componentInstance;
    spyOn(store, 'dispatch');
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'ng-household-expenses'`, () => {
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('ng-household-expenses');
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dipsatch ResizeWindow action', () => {
    const width = window.innerWidth;
    const height = window.innerHeight;
    const expected = new LayoutActions.ResizeWindow({
      width: width,
      height: height
    });

    window.dispatchEvent(new Event('resize'));

    expect(store.dispatch).toHaveBeenLastCalledWith(expected);
  });

  it('should dispatch Toggle Sidebar action', () => {
    const expected = new LayoutActions.ToggleSidebar();

    component.onSidebarToggled();

    expect(store.dispatch).toHaveBeenLastCalledWith(expected);
  });

  it('should dispatch Logout action', () => {
    const expected = new AuthActions.Logout();

    component.onLogout();

    expect(store.dispatch).toHaveBeenLastCalledWith(expected);
  });
});
