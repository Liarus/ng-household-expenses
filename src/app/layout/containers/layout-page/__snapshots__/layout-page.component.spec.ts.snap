// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutPageComponent should match snapshot 1`] = `
<app-layout-page
  isSidebarExpanded$={[Function Store]}
  menuItems$={[Function Store]}
  store={[Function MockStore]}
  title={[Function String]}
  user$={[Function Store]}
>
  <app-toolbar
    ng-reflect-user="[object Object]"
  >
    <mat-toolbar
      class="mat-toolbar mat-primary mat-toolbar-single-row"
      color="primary"
      fxflex=""
      ng-reflect-color="primary"
    >
      <div
        fxflex=""
        fxlayout="row"
      >
        <button
          class="app-toolbar-btn-toggle mat-icon-button _mat-animation-noopable"
          id="toolbar-btn-toggle"
          mat-icon-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon material-icons"
              role="img"
            >
              menu
            </mat-icon>
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <span
          fxflex=""
        />
        
        <div
          class="ng-star-inserted"
        >
          <span
            class="app-toolbar-user-name"
            fxshow.xs="false"
          >
            UserName
          </span>
        </div>
        <button
          class="mat-icon-button _mat-animation-noopable"
          id="toolbar-btn-logout"
          mat-icon-button=""
        >
          <span
            class="mat-button-wrapper"
          >
            <mat-icon
              aria-hidden="true"
              class="mat-icon material-icons"
              role="img"
            >
              exit_to_app
            </mat-icon>
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="false"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
      </div>
    </mat-toolbar>
  </app-toolbar><app-sidebar
    ng-reflect-is-expanded="true"
    ng-reflect-menu-items="[object Object],[object Object"
  >
    <mat-sidenav-container
      class="mat-drawer-container mat-sidenav-container"
      fxflexfill=""
      fxlayout="column"
    >
      
      <div
        class="mat-drawer-backdrop ng-star-inserted"
      />
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav
        class="mat-drawer mat-sidenav ng-tns-c16-0 ng-trigger ng-trigger-transform mat-drawer-side ng-star-inserted"
        fxlayout="column"
        mode="side"
        ng-reflect-mode="side"
        ng-reflect-opened="true"
        style="transform:none;visibility:visible;0:transform;1:visibility;"
        tabindex="-1"
      >
        <div
          class="mat-drawer-inner-container"
        >
          <mat-nav-list
            class="app-nav-list mat-nav-list mat-list-base app-nav-list-expanded"
            ng-reflect-klass="app-nav-list"
            ng-reflect-ng-class="app-nav-list-expanded "
            role="navigation"
          >
            
            <a
              class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
              href="/1"
              mat-list-item=""
              ng-reflect-router-link="/1"
              style=""
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="false"
                  ng-reflect-trigger="http://localhost/1"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-list-icon mat-icon material-icons"
                  mat-list-icon=""
                  role="img"
                >
                  home
                </mat-icon>
                <div
                  class="mat-list-text"
                >
                  <p
                    class="mat-line"
                    matline=""
                  >
                    menuTitle1
                  </p>
                </div>
              </div>
            </a>
            <a
              class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
              href="/2"
              mat-list-item=""
              ng-reflect-router-link="/2"
              style=""
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="false"
                  ng-reflect-trigger="http://localhost/2"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-list-icon mat-icon material-icons"
                  mat-list-icon=""
                  role="img"
                >
                  home
                </mat-icon>
                <div
                  class="mat-list-text"
                >
                  <p
                    class="mat-line"
                    matline=""
                  >
                    menuTitle2
                  </p>
                </div>
              </div>
            </a>
            <a
              class="mat-list-item mat-list-item-avatar mat-list-item-with-avatar ng-star-inserted"
              href="/3"
              mat-list-item=""
              ng-reflect-router-link="/3"
              style=""
            >
              <div
                class="mat-list-item-content"
              >
                <div
                  class="mat-list-item-ripple mat-ripple"
                  mat-ripple=""
                  ng-reflect-disabled="false"
                  ng-reflect-trigger="http://localhost/3"
                />
                <mat-icon
                  aria-hidden="true"
                  class="mat-list-icon mat-icon material-icons"
                  mat-list-icon=""
                  role="img"
                >
                  home
                </mat-icon>
                <div
                  class="mat-list-text"
                >
                  <p
                    class="mat-line"
                    matline=""
                  >
                    menuTitle3
                  </p>
                </div>
              </div>
            </a>
          </mat-nav-list>
        </div>
      </mat-sidenav>
      <div
        class="cdk-visually-hidden cdk-focus-trap-anchor"
      />
      <mat-sidenav-content
        class="app-sidenav mat-drawer-content mat-sidenav-content app-sidenav-content-expanded"
        ng-reflect-klass="app-sidenav"
        ng-reflect-ng-class="app-sidenav-content-expanded"
      >
        <router-outlet />
      </mat-sidenav-content>
      
    </mat-sidenav-container>
  </app-sidebar>
</app-layout-page>
`;
